src/conn.c

```diff
@@ -737,6 +737,13 @@ _makeTLSConn(natsConnection *nc)
                 SSL_set_verify(ssl, SSL_VERIFY_PEER, _collectSSLErr);
         }
     }
+#if defined(NATS_USE_OPENSSL_1_1)
+    // add the host name in the SNI extension
+    if ((s == NATS_OK) && (nc->cur != NULL) && (!SSL_set_tlsext_host_name(ssl, nc->cur->url->host)))
```

Comment:
This makes test `SSLVerifyHostname` fail if the library/tests are build with `NATS_FORCE_HOST_VERIFICATION=OFF`. So I would say that at the very least we need to add the condition of `NATS_FORCE_HOST_VERIFICATION` enabled as condition to call this function.

But I would also like to see a test that demonstrates why this is necessary for the test to succeed (that is, without this new line it would fail). The presence of it seem to not fail the existing tests (well, except when name verification is not forced).

Reply:
I take back that this change makes the test fails. It fails without your changes as long as the force host verification is disabled it seems. I am investigating.

Reply:
Ok, but still we would need a test that shows that it works ;-)

Reply:
TBH I have no idea how to test this ;)

Reply:
there is nothing that tests the SSL config at this point. It is like a blackbox. I can tell you that it works because I tried it out on a specific setup but that's it.

Reply:
I am assuming that you know it works because you tried in your env but don't know how to write a test about it? I mean, is it a matter of having the server's TLS configure a certain way? Any information you could give.